### ISSUE TITLE
Allow for creation of address groups

### ISSUE THREAD

**Issue by oliver-graetz** [ethanliu/location-switcher-addon#18]
When I have four sites for a project, and want to switch between all of them, then currently I need to create twelve entries in the configuration, three for each of the sites with the three others as destinations. This is really tedious and also prone to errors. It'd be much more simple to be able to create groups in the config and then just enter the four base addresses, and then choose between cycling or destination selection.

Also, it would be good if additionally to source and destination, each of the entries could have a name to display instead of its address.

**Comment by ethanliu** [comment]
group seems a good solution for multiple destinations

**Comment by MurzNN** [comment]
Yeah, if we have more than 2 domains for a developing project, it's usually pretty complicated to configure the switching right. Usually, we have a local domain, dev, test, uat and prod domains, so 5 domains for one project.

An example of the interface with groups you can look up in the https://addons.mozilla.org/en-US/firefox/addon/domain-master-and-ping/ extension and https://chrome.google.com/webstore/detail/domain-switcher/lbehdhpgigdlinfkidifkbhjnaglfojc for Chrome.

But Location Switcher extension has a very cool feature with assigning custom icons to projects, so I'm still fan of it! :)

**Comment by ethanliu** [comment]
The currently challenge would be the data layer, since it use the local storage that's a simple key/value pair structure, it's less flexible to do things like project/group/tag, regular expression stuff. considering replace with IndexedDB or Sqlite

**Comment by MurzNN** [comment]
Found one more open-source extension https://addons.mozilla.org/en-US/firefox/addon/mpopp75-site-switcher/ that supports groups of domains per environment - hope it can help with the implementation.

**Comment by ethanliu** [comment]
while no any promise yet, I'll probably rebuild with IndexedDB which is more future proof.
